### ISSUE TITLE
 "Taks 2 : Audio Transcription Tool with SpeechRecognition and Wav2Vec2"


### ISSUE THREAD

**Issue by Ycrakeshreddy** [Ycrakeshreddy/rakeshreddy21#1]
1✅ Task 2: Speech-to-Text System using Pre-trained Models

🔧 Objective:

Build a basic Speech-to-Text (STT) system using pre-trained models and libraries like SpeechRecognition or Wav2Vec.

---

🧠 Approach:

We’ll provide two implementations:

1. Using SpeechRecognition (simple and easy setup).

2. Using Facebook’s Wav2Vec 2.0 model from Hugging Face for high accuracy.

---

🔹 Option 1: Using SpeechRecognition Library

📜 Python Script:

# speech_to_text_simple.py

import speech_recognition as sr

def recognize_from_audio(file_path):
    recognizer = sr.Recognizer()
    with sr.AudioFile(file_path) as source:
        audio_data = recognizer.record(source)
        try:
            text = recognizer.recognize_google(audio_data)
            print("Transcribed Text:\n", text)
        except sr.UnknownValueError:
            print("Could not understand audio.")
        except sr.RequestError as e:
            print("API request error:", e)

if __name__ == "__main__":
    audio_file = "sample.wav"  # Use your own WAV audio file here
    recognize_from_audio(audio_file)

📦 Requirements:

pip install SpeechRecognition

> 🔊 Note: Only WAV files are supported by default. Convert other formats using tools like Audacity or ffmpeg.

---

🔹 Option 2: Using Wav2Vec 2.0 from Hugging Face

📜 Python Script:

# speech_to_text_wav2vec.py

import torch
import librosa
from transformers import Wav2Vec2ForCTC, Wav2Vec2Processor

def transcribe_audio(file_path):
    print("Loading model...")
    processor = Wav2Vec2Processor.from_pretrained("facebook/wav2vec2-base-960h")
    model = Wav2Vec2ForCTC.from_pretrained("facebook/wav2vec2-base-960h")

    print("Loading audio...")
    audio, rate = librosa.load(file_path, sr=16000)
    input_values = processor(audio, return_tensors="pt", sampling_rate=16000).input_values

    print("Generating prediction...")
    with torch.no_grad():
        logits = model(input_values).logits

    predicted_ids = torch.argmax(logits, dim=-1)
    transcription = processor.batch_decode(predicted_ids)[0]
    print("Transcription:\n", transcription)

if __name__ == "__main__":
    transcribe_audio("sample.wav")  # Replace with your WAV file

📦 Requirements:

pip install torch librosa transformers

---

💡 Sample Audio Source:

You can create or download a short .wav file with clear speech for testing.
Ensure the duration is under 30 seconds and use 16kHz sampling rate.

---

📝 Deliverables:

✔️ Two Python scripts: one simple, one advanced.

✔️ Capable of transcribing short audio clips into text.

✔️ Compatible with .wav files.
